tests/rest/devices.spec.js

```diff
@@ -0,0 +1,211 @@
+
+          res.body.should.be.a('array');
+          res.body.length.should.be.eql(4);
+
+          res.body[2].should.be.a('object');
+          res.body[2].should.have.property('psk_id');
+          res.body[2].psk_id.should.be.eql('cHNraWQy');
+
+          done();
+        });
+    });
+  });
+
+  describe('GET /devices:name', function() {
+    it('should return a single object', (done) => {
+      chai.request(server)
+        .get('/devices/ABC')
+        .end((err, res) => {
+          should.not.exist(err);
+          res.should.have.status(200);
+          res.should.have.header('content-type', 'application/json');
+
+          res.body.should.be.a('object');
+          res.body.should.have.property('psk_id');
+          res.body.psk_id.should.be.eql('cHNraWQx');
+
+          done();
+        });
+    });
+
+    it('where \'name\' is non-existing should return 404', (done) => {
+      chai.request(server)
+        .get('/devices/non-existing')
+        .end((err, res) => {
+          res.should.have.status(404);
+          done();
+        });
+    });
+  });
+
+  describe('PUT /devices:name', function() {
+    it('should return 201', (done) => {
+      chai.request(server)
+        .put('/devices/ABC')
+        .set('Content-Type', 'application/json')
+        .send('{"psk":"cHNrMQ==","psk_id":"cHNraWQa"}')
+        .end((err, res) => {
+          should.not.exist(err);
+          res.should.have.status(201);
+
+          chai.request(server)
+            .get('/devices/ABC')
+            .end((error, response) => {
+                should.not.exist(error);
+                response.should.have.status(200);
+
+                response.should.have.header('content-type', 'application/json');
+                response.body.should.be.a('object');
+                response.body.should.have.property('psk_id');
+                response.body.psk_id.should.be.eql('cHNraWQa');
+
+                done();
+            });
+        });
+    });
+
+    it('where \'name\' is non-existing should return 400', (done) => {
+      chai.request(server)
+        .put('/devices/non-existing')
+        .set('Content-Type', 'application/json')
+        .send('{"psk":"cHNrMQ==","psk_id":"cHNraWQa"}')
+        .end((err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+  });
+
+  describe('DELETE /devices:name', function() {
+    it('should return 200', (done) => {
+      chai.request(server)
+        .delete('/devices/GHI')
+        .end((err, res) => {
+          should.not.exist(err);
+          res.should.have.status(200);
+
+          chai.request(server)
+            .get('/devices/GHI')
+            .end((error, response) => {
+                response.should.have.status(404);
+                done();
+            });
+        });
+    });
+
+    it('where \'name\' is non-existing should return 404', (done) => {
+      chai.request(server)
+        .delete('/devices/non-existing')
```

Comment:
This could actually repeat the previous request, since it should be deleted and non-exiting.